src/routes/events/containers/EventsPage.js

```diff
@@ -26,6 +29,43 @@ class EventsPage extends React.Component {
     return `/${this.props.location}/${this.props.language}/events`
   }
 
+  mapLanguageToUrl = (language, id) =>
+    id ? `/${this.props.location}/${language}/events/${id}`
+      : `/${this.props.location}/${language}/events`
+
+  componentDidMount () {
+    if (this.props.path && this.props.events) {
+      // specific event
+      const event = this.props.events.find(
+        (event) => event.id.toString() === this.props.path.replace('/', '')
+      )
+      this.props.dispatch(setLanguageChangeUrls(this.mapLanguageToUrl, this.props.languages, event.availableLanguages))
+    } else {
+      // all events
+      this.props.dispatch(setLanguageChangeUrls(this.mapLanguageToUrl, this.props.languages))
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (nextProps.events !== this.props.events || nextProps.path !== this.props.path) {
```

Comment:
Simplify this like componentDidMount

Reply:
simplified it a little bit, but if we would simplify this further, we would have to dispatch twice